feat: add arguments optional lab to full stack

majestic-owl448 opened the pull request:
Checklist:

<!-- Please follow this checklist and put an x in each of the boxes, like this: [x]. It will ensure that our team takes your pull request seriously. -->

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or Gitpod.

<!--If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.-->

Closes https://github.com/freeCodeCamp/CurriculumExpansion/issues/927

<!-- Feel free to add any additional description of changes below this line -->

majestic-owl448 committed:
created files

majestic-owl448 committed:
fin

ilenia-magoni commented:
I forgot the intro, turning back into draft until I add it

majestic-owl448 committed:
intro

majestic-owl448 committed:
capitalization

ilenia-magoni commented on curriculum/challenges/english/25-front-end-development/lab-optional-arguments-sum-function/a97fd23d9b809dac9921074f.md:
```diff
@@ -0,0 +1,120 @@
+
+# --description--
+
+In this lab you will build a function that sum its arguments, where the arguments can be optional.
```
Maybe it is more clear with
```suggestion
In this lab you will build a function that accepts up to two arguments, and sum them, but if there is only one argument returns a function that waits for the second number to sum.
```

what do you think?

majestic-owl448 committed:
Merge main and apply curriculum structure migration

majestic-owl448 committed:
Update challenge description for optional arguments sum function

Clarify that the function accepts up to two arguments and returns a function when only one argument is provided.

majestic-owl448 committed:
Update description in index.md and intro.json for optional arguments sum function

Apply the same description update to the client-side files to maintain consistency across all references to this lab.
